### PR TITLE
Tellor Oracle Tutorial

### PR DESCRIPTION
**Pull request by RaynHarr**
Tutorial for getting set up with the Tellor oracle.

## Description
Submission for 'How to Set Up the Tellor Oracle' to be added to the developer tutorials page.

### CODE REVIEW

**Comment by wackerow** [comment]
@RaynHarr Hey, thanks a lot for the tutorial PR! Bear with us, will take a closer look within a week or two and get back to you with feedback 🙂

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 1]
```diff
@@ -0,0 +1,145 @@
+---
+title: How to Set Up Tellor as your Oracle
+description: A guide to get started with integrating the Tellor oracle into your protocol
+author: "Tellor"
+lang: en
+sidebar: true
+tags: ["solidity", "smart contracts", "pricefeeds", "oracles"]
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:7]: would use "price feeds"

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+
+## UsingTellor
+First thing you'll want to do is install the basic tools necessary for using Tellor as your oracle. Use [this package](https://github.com/tellor-io/usingtellor) to install the Tellor User Contracts to test the implementation of Tellor in your contracts:
+
+`npm install usingtellor`
+
+Once installed this will allow your contracts to inherit the functions from the contract 'UsingTellor'. 
+
+
+Great! Now that you've got the tools ready, let's go through a simple excercise where we request the Bitcoin price:
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:37]: - "exercise"
- lowercase "bitcoin" (for the currency)

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+## Available Tellor Functions:
+
+Below is a group of basic functions that come with inheriting the UsingTellor contract. These will allow you to utilize Tellor in a more robust fashion than the example above.
+
+Children contracts have access to the following functions:
+
+```solidity
+    /**
+    * @dev Retreive value from oracle based on requestId/timestamp
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:76]: - "Retrieve" 
- (3 other instances of this typo noted below)

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    * @param _requestId being requested
+    * @param _timestamp to retreive data/value from
+    * @return uint value for requestId/timestamp submitted
+    */
+    function retrieveData(uint256 _requestId, uint256 _timestamp) public view returns(uint256);
+
+    /**
+    * @dev Gets if the mined value for the specified requestId/_timestamp is currently under dispute
+    * @param _requestId to looku p
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:85]: "look up"

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+    * @param _timestamp is the timestamp to look up miners for
+    * @return bool true if requestId/timestamp is under dispute
+    */
+    function isInDispute(uint256 _requestId, uint256 _timestamp) public view returns(bool);
+
+    /**
+    * @dev Counts the number of values that have been submited for the request
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:92]: "submitted"

File: src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+
+Rinkeby: [`0x88df592f8eb5d7bd38bfef7deb0fbc02cf3778a0`](https://rinkeby.etherscan.io/address/0x20374E579832859f180536A69093A126Db1c8aE9#code)
+
+Kovan: [`0x20374E579832859f180536A69093A126Db1c8aE9`](https://kovan.etherscan.io/address/0x20374E579832859f180536A69093A126Db1c8aE9#code)
+
+Ropsten: [`0x20374E579832859f180536A69093A126Db1c8aE9`](https://ropsten.etherscan.io/address/0x20374E579832859f180536A69093A126Db1c8aE9#code)
+
+Goerli: [`0x20374E579832859f180536A69093A126Db1c8aE9`](https://goerli.etherscan.io/address/0x20374E579832859f180536A69093A126Db1c8aE9#code)
+
+<br>
```
Comment (wackerow) [inline src/content/developers/tutorials/how-to-use-tellor-as-your-oracle/index.md:139]: Can remove `<br>` in the markdown

**Review by wackerow — CHANGES_REQUESTED** [review]
@RaynHarr Thanks again for posting this... noted a few small adjustments, but otherwise looked pretty good. 

I see it's labeled as beginning, just curious if we should include a bit more guidance as a beginner tutorial about how to go about using the Tellor functions listed. I feel this currently assumes a certain level of understanding, which is fine, but if there are pre-requisites maybe we should note those at the beginning and label this intermediate or advanced. 

Thoughts? Then we'll get this merged =)

**Comment by RaynHarr** [comment]
Hey Paul! 

Sorry I missed this, didn’t realize how fast you guys were! 

Absolutely, we’ll look at it today and get back. 

Thanks!

-Ryan

**Comment by RaynHarr** [comment]
@wackerow Great points you brought up. In an attempt to keep it beginner-friendly, we went ahead and removed the extensive list of functions and gave a preliminary list of assumptions. Let me know if this works better, and if you think it's still wise to move it to intermediate, then that's cool with us too. Also, thanks for reminding me how to spell retrieved -_-

**Review by wackerow — APPROVED** [review]
@RaynHarr Looks good 👍🏻 Thanks for the updates, will get this pulled in

**Comment by wackerow** [comment]
@all-contributors please add @RaynHarr for content

**Comment by RaynHarr** [comment]
Woohoo! Thanks so much @wackerow and everyone! Much thanks to @tkernell as well for the guidance and collaboration.

**Pull request merged**